warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctions.java

```diff
@@ -310,6 +307,69 @@ public static boolean excludeRegex(Iterable<?> values, String regex) {
         return FunctionalSet.unmodifiableSet(matches);
     }
     
+    private static class NaturalComparator<T extends Comparable<T>> implements Comparator<T> {
+        public int compare(T a, T b) {
+            
+            return a.compareTo(b);
+        }
+    }
+    
+    /**
+     * We don't have access, at this layer, to the type information. We do have the typed value, so we must expect there to be type T that matches this. If they
+     * do not the comparison will result in a class cast exception, through which we we can attempt to create the corresponding hard type.
+     * 
+     * @param fieldValue
+     *            ValueTuple that contains the raw field value
+     * @param lowerBound
+     *            lower bound, expected to be the same type as the raw field value
+     * @param upperBound
+     *            upper bound, expected to be the same field value.
+     * @param <T>
+     *            Generic type of the bounds arguments.
+     * @return
+     */
+    public static <T> Collection<ValueTuple> getWithinInclusiveRange(ValueTuple fieldValue, T lowerBound, T upperBound) {
+        try {
+            if (Range.between(lowerBound, upperBound, new NaturalComparator()).contains((T) ValueTuple.getNormalizedValue(fieldValue))) {
+                return FunctionalSet.singleton(getHitTerm(fieldValue));
+            }
+        } catch (ClassCastException e) {
+            if (fieldValue.getNormalizedValue() instanceof BigDecimal) {
```

Comment:
I would love input here. The expectation is that if I have a range that can be converted ( if a string) we allow that range execution to occur, otherwise the query will produce an exception by way of the class cast.

Reply:
Only just started looking at this, but my initial thoughts are that it might be worth considering adding functionality to the FunctionNormalizationRebuildingVisitor or some other FunctionRangeHandlingVisitor, since there you have access to the normalization information and can determine rangeablitiy, as it were, maybe some validation that the lowerBound is indeed less than the upperBound, etc., and somehow pass down the type information you need here.  Also that way we could fail up in Planning if we are not dealing with a rangeable type, and other functions we might want to add range functionality to will inherit the goodness.  Maybe that's hokey.  Maybe I've misinterpreted this idea.  Idk, It's after 9 on a Friday.